Source/Core/VideoCommon/ShaderCache.cpp

```diff
@@ -608,14 +608,10 @@ AbstractPipelineConfig ShaderCache::GetGXPipelineConfig(
 static GXPipelineUid ApplyDriverBugs(const GXPipelineUid& in)
 {
   GXPipelineUid out;
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wclass-memaccess"
-#endif
-  memcpy(&out, &in, sizeof(out));  // copy padding
-#ifdef __GNUC__
-#pragma GCC diagnostic pop
-#endif
+  // TODO: static_assert(std::is_trivially_copyable_v<GXPipelineUid>);
+  // GXPipelineUid is not trivially copyable because of BitField memes, but we can
```

Comment:
I was being a bit silly with that comment, but now I can see that it was pretty unhelpful to be so nonspecific.  The reason GXPipelineUid and GXUberPipelineUid are not trivially copyable is due to them including RasterizationState and BlendingState, not because of classic BitField's deleted copy constructor.  The semantics of the classical BitField still make my head spin, but I think the difference is BitField *deletes* copy assignment, while RasterizationState/BlendingState have *user-defined* copy assignment.  [At your suggestion](https://github.com/dolphin-emu/dolphin/pull/10610#issuecomment-1114132249), PR #10848 removes these, and should make the classes trivially copyable.  The BitFieldView changes should also make these classes copy assignable, making all these memcpys pointless.